File: khalti-android/src/main/java/com/khalti/checkOut/EBanking/deepLinkReceiver/DeepLink.java
package com.khalti.checkOut.EBanking.deepLinkReceiver;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;

import java.util.HashMap;

import com.khalti.R;
import com.khalti.checkOut.helper.Config;
import com.khalti.utils.EmptyUtil;
import com.khalti.utils.FileStorageUtil;
import com.khalti.utils.JsonUtil;

public class DeepLink extends AppCompatActivity implements DeepLinkContract.View {

    private DeepLinkContract.Listener listener;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.empty_activity);

        listener = new DeepLinkPresenter(this);
        listener.receiveEBankingData();
    }

    @Override
    public HashMap<String, Object> receiveEBankingData() {
        Bundle bundle = getIntent().getExtras();
        if (EmptyUtil.isNotNull(bundle)) {
            return JsonUtil.getEBankingData(bundle.getString("data"));
        }
        return null;
    }

    @Override
    public Config getConfigFromFile() {
        return (Config) FileStorageUtil.readFromFile(this, "khalti_config");
    }

    @Override
    public void closeDeepLink() {
        finish();
    }

    @Override
    public void setListener(DeepLinkContract.Listener listener) {
        this.listener = listener;
    }
}

File: khalti-android/src/main/java/com/khalti/checkOut/ebanking/deepLinkReceiver/DeepLink.kt
package com.khalti.checkOut.ebanking.deepLinkReceiver

import android.content.ComponentName
import android.content.ServiceConnection
import android.os.Bundle
import android.os.IBinder
import androidx.appcompat.app.AppCompatActivity

import java.util.HashMap

import com.khalti.R
import com.khalti.checkOut.helper.Config
import com.khalti.checkOut.service.ConfigService
import com.khalti.checkOut.service.ConfigServiceComm
import com.khalti.utils.EmptyUtil
import com.khalti.utils.FileStorageUtil
import com.khalti.utils.JsonUtil

class DeepLink : AppCompatActivity(), DeepLinkContract.View {

    private lateinit var presenter: DeepLinkContract.Presenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.empty_activity)

        presenter = DeepLinkPresenter(this)
        presenter.onCreate()
    }

    override val configFromFile: Config
        get() = FileStorageUtil.readFromFile(this, "khalti_config") as Config

    override val configFromService: Config?
        get() = ConfigService.getConfig()

    override fun receiveEBankingData(): Map<String, Any>? {
        val bundle = intent.extras
        return if (EmptyUtil.isNotNull(bundle)) {
            JsonUtil.getEBankingData(bundle!!.getString("data"))
        } else null
    }

    override fun closeDeepLink() {
        finish()
    }

    override fun setPresenter(presenter: DeepLinkContract.Presenter) {
        this.presenter = presenter
    }
}